MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -863,6 +867,103 @@ public PersonnelStatus getStatus() {
         return status;
     }
 
+    /**
+     * This is used to change the person's PersonnelStatus
+     * @param campaign the campaign the person is part of
+     * @param status the person's new PersonnelStatus
+     */
+    public void changeStatus(Campaign campaign, PersonnelStatus status) {
+        if (status == getStatus()) { // no change means we don't need to process anything
+            return;
+        } else if (getStatus().isKIA()) {
+            // remove date of death for resurrection
+            setDateOfDeath(null);
+        }
+
+        switch (status) {
+            case ACTIVE:
+                if (getStatus().isMIA()) {
+                    ServiceLogger.recoveredMia(this, campaign.getDate());
+                } else if (getStatus().isDead()) {
+                    ServiceLogger.resurrected(this, campaign.getDate());
+                } else {
```

Comment:
Should we make this explicit? Might look weird if the person is in a status where it doesn't make sense for them to be re-hired.

Reply:
The above handle all of the potential statuses that would not be a rehire, plus this would make the most sense for any status being changed to active that is not MIA or a version of death. (unless I'm missing a potential future addition, but this should be a solid future proofed method to do so)